src/types/index.ts

```diff
@@ -1,10 +1,88 @@
-import { ManagedIdentityTokenSource } from "../tokensource";
-
 export * from "./activityTypes";
 export * from "./durableClientTypes";
 export * from "./entityTypes";
 export * from "./orchestrationTypes";
 export * from "./taskTypes";
 
+/**
+ * Token Source implementation for [Azure Managed Identities](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview).
+ *
+ * @example Get a list of Azure Subscriptions by calling the Azure Resource Manager HTTP API.
+ * ```javascript
+ * const df = require("durable-functions");
+ *
+ * df.app.orchestration(function* (context) {
+ *   return yield context.df.callHttp({
+ *       method: "GET",
+ *       url: "https://management.azure.com/subscriptions?api-version=2019-06-01",
+ *       tokenSource: df.ManagedIdentityTokenSource("https://management.core.windows.net"),
+ *   });
+ * });
+ * ```
+ */
+export declare class ManagedIdentityTokenSource {
+    /**
+     * Returns a `ManagedIdentityTokenSource` object.
+     * @param resource The Azure Active Directory resource identifier of the web API being invoked.
+     */
+    constructor(resource: string);
+
+    /**
+     * The Azure Active Directory resource identifier of the web API being invoked.
+     * For example, `https://management.core.windows.net/` or `https://graph.microsoft.com/`.
+     */
+    readonly resource: string;
+
+    /** @hidden */
+    kind: string;
+}
+
 // Over time we will likely add more implementations
 export type TokenSource = ManagedIdentityTokenSource;
+
+/**
+ * Defines retry policies that can be passed as parameters to various
+ * operations.
+ */
+export declare class RetryOptions {
+    /**
+     * Creates a new instance of RetryOptions with the supplied first retry and
+     * max attempts.
+     * @param firstRetryIntervalInMilliseconds The first retry interval (ms). Must be greater than 0.
+     * @param maxNumberOfAttempts The maximum number of attempts.
+     */
+    constructor(firstRetryIntervalInMilliseconds: number, maxNumberOfAttempts: number);
+    /**
+     * The retry backoff coefficinet
+     */
+    backoffCoefficient: number;
+    /**
+     *  The max retry interval (ms).
+     */
+    maxRetryIntervalInMilliseconds: number;
+    /**
+     * The timeout for retries (ms).
+     */
+    retryTimeoutInMilliseconds: number;
+    /**
+     * The first retry interval (ms). Must be greater than 0.
+     */
+    readonly firstRetryIntervalInMilliseconds: number;
+    /**
+     * The maximum number of attempts.
+     */
+    readonly maxNumberOfAttempts: number;
+}
+
+/**
+ * A specfic error thrown when a scheduled activity or suborchestrator has failed.
+ * This error can be checked for via `instanceof` guards to catch only exceptions thrown
+ * by the DurableJS library.
+ */
+export declare class DurableError extends Error {
+    /**
+     * Constructs a `DurableError` instance.
+     * @param message error message.
+     */
+    constructor(message?: string);
```

Comment:
`message?: string` is slightly different than `message: string | undefined`, which is how the class is defined in `src`. I'd prefer to keep them consistent

Reply:
How is it different?

Reply:
With `message?: string`, its easier to forget to pass `message`. So for example if you want people to pass a message 99% of the time, you'd use `message: string | undefined` to remind people. If `message` is unimportant and used less often, you'd use `message?: string` for the sake of convenience.

I don't really care which one we choose in this case, but I think the types and the src file should be consistent

Reply:
Ok fair enough. Since it was `| undefined` in the `src`, I've changed it to be the same in the `types` too.